pypolestar/polestar.py

```
@@ -109,9 +108,11 @@ def get_car_information(self, vin: str) -> CarInformationData | None:
             KeyError: If the VIN doesn't exist
             ValueError: If data conversion fails
         """
+
         if vin not in self.available_vins:
             raise KeyError(vin)
-        if data := self.data_by_vin[vin].get(CAR_INFO_DATA, {}).get("data"):
+
+        if data := self.data_by_vin[vin].get(CAR_INFO_DATA):
```

maybe test if data is not None instead of relying on the exception ? 

Furthermore the error message could use the VIN number in order to assist in debugging

---

Perhaps something like 0cb938309d19a91fb5618aa949ded7581cbcc6ec  ?

---

That is great! But I was more concerned with the get() which might return None

---

Do you think we should raise an exception if no data is available instead of returning None?

---

Unfortunately the API is down more then we would expect. Returning None is fine, and should be processed further ahead in a retry logic.

---

We should state that callers should use `get_ev_data` for polling (this will never happen more often than `next_update_delay`, current X=5), then `get_car_XXX` to get the latest available data. All other methods should be considered internal only.

There is currently a global lock for updates, I should probably make this per VIN.